### DISCUSSION TITLE
Is it possible for all devices that have SGX to join a network? Or only nodes hosted on Azure?

### DISCUSSION THREAD

**Discussion by Dutchprogrammer** [microsoft/CCF#3257] [Q&A] [answered]
Is it possible for all devices with SGX to join a network? We tried to join a network with a laptop running Ubuntu 20.04.3 with SGX. However, when joining the network (hosted on Azure) it keeps failing with the error message std::logic_error: An error occurred when creating CCF node: InternalError

On Azure, the same configuration CCF joins the network(also on Azure) without any errors. It seems that you cannot join a network out of Azure.

**Comment by achamayou** [comment, marked as answer]
@Dutchprogrammer it is possible, CCF relies on Open Enclave for [remote attestation](https://github.com/openenclave/openenclave/blob/168cadc2f47b13ce47fecbb46a586004f74392f7/samples/attestation/README.md), so you will need to set up your own Provisioning Certificate Caching Service, and install a slightly different DCAP configuration than in Azure. The steps you need to take are documented [here](https://github.com/openenclave/openenclave/blob/28f682e9a82be688e486499ef721ac955e7bfa71/docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md).

**Reply by Dutchprogrammer** [reply]
Thank you for your quick reply! I will look into it. Thanks

**Reply by achamayou** [reply]
I should probably clarify that the CCF team hasn't used that setup before, and has no immediate plan to do so. CCF strictly uses Open Enclave's public API however, so as long as you can get Open Enclave's samples working with a local setup, we would expect CCF to work as well. Should that turn out not to be the case, please let us know.

**Comment by achamayou** [comment]
CCF no longer supports SGX.
